keep/api/core/db.py

```
@@ -4482,3 +4487,43 @@ def get_resource_ids_by_resource_type(
         # Execute the query and return results
         result = session.exec(query)
         return result.all()
+
+
+def get_last_alert_by_fingerprint(
+    tenant_id: str, fingerprint: str, session: Optional[Session] = None
```

Ruff

`Optional` may be undefined, or defined from star imports

(F405)